src/artemis/devices/unit_tests/test_oav_centring.py

```diff
@@ -1,32 +1,335 @@
+            diff_at_x_pos,
+            mid,
+        ) = rotate_pin_and_collect_positional_data(mock_oav, mock_smargon, 6)
+
+    with pytest.raises(OAVError_WaveformAllZero):
+        RE = RunEngine()
+        RE(
+            fake_run(mock_oav, mock_smargon),
+        )
+
+"""
+
+"""
+def test_all_zero_waveform(fake_mv, mock_oav: OAV, mock_smargon: I03Smargon):
+
+    x = np.zeros(1024)
+
+    def fake_run(mock_oav: OAV, mock_smargon: I03Smargon):
+        mock_smargon.wait_for_connection = do_nothing
+        yield from bps.abs_set(mock_smargon.omega, 0)
+        yield from bps.abs_set(mock_oav.mxsc.top, x)
+        yield from bps.abs_set(mock_oav.mxsc.bottom, x)
+        yield from bps.abs_set(mock_oav.mxsc.tip_x, 0)
+        yield from bps.abs_set(mock_oav.mxsc.tip_y, 0)
+
+        (
+            x_pos,
+            y_pos,
+            diff_at_x_pos,
+            mid,
+        ) = rotate_pin_and_collect_values(mock_oav, mock_smargon, 6)
+
+    with pytest.raises(OAVError_WaveformAllZero):
+        RE = RunEngine()
+        RE(
+            fake_run(mock_oav, mock_smargon),
+        )
+
+
+"""
+
+
+def test_filter_rotation_data():
+    x_positions = np.array([400, 450, 7, 500])
+    y_positions = np.array([400, 450, 7, 500])
+    widths = np.array([400, 450, 7, 500])
+    omegas = np.array([400, 450, 7, 500])
+
+    (
+        filtered_x,
+        filtered_y,
+        filtered_widths,
+        filtered_omegas,
+    ) = filter_rotation_data(x_positions, y_positions, widths, omegas)
+
+    assert filtered_x[2] == 500
+    assert filtered_omegas[2] == 500
+
+
+@pytest.mark.parametrize(
+    "max_tip_distance, tip_x, x, expected_return",
+    [
+        (180, 400, 600, 580),
+        (180, 400, 450, 450),
+    ],
+)
+def test_keep_x_within_bounds(max_tip_distance, tip_x, x, expected_return):
+    assert check_x_within_bounds(max_tip_distance, tip_x, x) == expected_return
+
+
+"""
+@pytest.mark.parametrize(
+    "h,v,omega,expected_values",
+    [
+        (0.0, 0.0, 0.0, np.array([0.0, 0.0, 0.0])),
+        (10, -5, 90, np.array([-10, 3.062e-16, -5])),
+        (100, -50, 40, np.array([-100, 38.302, -32.139])),
+        (10, 100, -4, np.array([-10, -99.756, -6.976])),
+    ],
+)
+def test_distance_from_beam_centre_to_motor_coords_returns_the_same_values_as_GDA(
+    h, v, omega, expected_values, mock_parameters: OAVParameters
+):
+
+    mock_parameters.zoom = 5.0
+    mock_parameters.load_microns_per_pixel()
+    results = camera_coordinates_to_xyz(
+        h,
+        v,
+        omega,
+        mock_parameters.micronsPerXPixel,
+        mock_parameters.micronsPerYPixel,
+    )
+    expected_values = expected_values * 1e-3
+    expected_values[0] /= mock_parameters.micronsPerXPixel
+    expected_values[1] /= mock_parameters.micronsPerYPixel
+    expected_values[2] /= mock_parameters.micronsPerYPixel
+    expected_values = np.around(expected_values, decimals=5)
+
+    assert np.array_equal(np.around(results, decimals=5), expected_values)
+"""
```

Comment:
Should: Can you either reinstate or remove this?

Reply:
I'll start writing writing the system tests now. Should be a seperate branch.